pymeasure/adapters/adapter.py

```diff
@@ -33,48 +37,135 @@ class Adapter:
 
     This class should only be inherited from.
 
-    :param preprocess_reply: optional callable used to preprocess strings
-        received from the instrument. The callable returns the processed string.
+    :param preprocess_reply: (deprecated) optional callable used to preprocess
+        strings received from the instrument. The callable returns the
+        processed string.
+    :param float query_delay: Time in s to wait after writing and before reading.
```

Comment:
I think this is a property of the `Instrument` and not an `Adapter` features.
It is also true that if I am using an Arduino based GPIB interface, it could happen that for some reason this adapter would require an extra delay which is not needed with NI GPIB adapter (this is a totally fictional example), but if this is the case, I would need to do the following:
- Implement a class `ArduinoGPIBAdapter(Adapter)` # This is need to use my custom adapter.
- Implement the extra delay in the _write method of `ArduinoGPIBAdapter`

So, in my opinion this parameter should not be added in the `Adapter`.

Reply:
The idea was, to have the delay in both places, instrument and adapter. Adding it to write would also wait between consecutive writes (without intermediate read). The current wait allows to wait only between write and read and further, you do not have to implement a new adapter.

Now is the question, whether we want to have this functionality in the adapter or require a custom adapter. 
I don't see many actual uses of this functionality, and thus I'm fine either way.
@bilderbuchi wanted to have this functionality.

We could remove this functionality (as you suggest) and open an issue asking users, whether they would deem it useful. If there someone would benefit from this feature, it's easy to add and does not break anything else.

Reply:
I was concerned about different delays for different interfaces (e.g. LAN vs serial), I think (I don't recollect clearly). But then, those would use the same adapter class probably. 
Maybe it is really best to, for simplicity, leave only one place for `query_delay` for the wait between a read and a write in a query -- probably Instrument is the proper place, I don't have a strong opinion on this.
An issue collecting user feedback on the needs w.r.t to this delay feature is a good idea!

Reply:
If you use an Instrument with an adapter requiring a delay, you know that the adapter requires the delay and you can invoke the Instrument with the appropriate delay. Consequently, the Adapter class does not have to know anything about delays.

As we have several methods writing and reading (`ask` and `binary_values`, maybe `wait_for_srq`), I propose to make a single `wait_till_read` method, invoked by them.

I guess, an adapter with a delay is an edge case. Therefore, I propose, that `Instrument` does not have a `query_delay` property. If an instrument (or adapter) requires a delay, the author (or user) shall subclass the `wait_till_read` method and implement waiting in this method, which does nothing (pass) in `Instrument`.

Reply:
The query delay functionality is now not in adapter anymore, but Instrument, so I'm pretty sure that this can be resolved.